java/code/src/com/suse/manager/maintenance/MaintenanceManager.java

```diff
@@ -695,6 +696,71 @@ public boolean isActionInMaintenanceWindow(Action action, MaintenanceSchedule sc
         return false;
     }
 
+    private Collection<CalendarComponent> getCalendarForNow(MaintenanceSchedule ms) {
+        return getScheduleEventsAtDate(new Date(), ms, parseCalendar(ms.getCalendarOpt().get()));
+    }
+
+    /**
+     * Check if system is in maintenance mode
+     *
+     * @param server the server to check
+     * @return true when the action is inside of a maintenance window, otherwise falsegg
+     */
+    public static boolean checkIfInMaintenanceMode(MinionServer server) {
+        if (server.getMaintenanceScheduleOpt().isEmpty()) {
+            return true;
+        }
+        else {
+            MaintenanceManager mm = MaintenanceManager.instance();
+            Collection<CalendarComponent> cc = mm.getCalendarForNow(server.getMaintenanceScheduleOpt().get());
+            return !cc.isEmpty();
+        }
```

Comment:
A proper way of doing this would be:

```suggestion
        return server.getMaintenanceScheduleOpt()
                .map(sched -> !mm.getCalendarForNow(sched).isEmpty())
                .orElse(true); 
```